### PR TITLE
Make code more consistent/pythonic

### PR DESCRIPTION
**Pull request by Grazfather**
In python empty strings and empty lists are considered False.
(bool([]) == False) and (bool("") == False).

This means that anything that does

    if len(l) > 0:

Can be replaced with

    if l:

I will work on other changes if you like these.

### CODE REVIEW

**Comment by Grazfather** [comment]
I wish I were working on features, but I can kind of do this kind of stuff mindlessly and it's something I've been doing to get ready for bed :)

File: gef.py [hunk 1]
```diff
@@ -2826,16 +2826,16 @@ def do_invoke(self, argv):
         with open(fname, "r") as f:
             p = re.compile(r"unknown_([a-f0-9]+)")
             for l in f.readlines():
-                l = l.rstrip()
-                if len(l.strip()) == 0 or l.strip().startswith("//"):
+                l = l.strip()
```
Comment (Grazfather) [inline gef.py]: I'll need to test this part out.

**Comment by hugsy** [comment]
I try to respect the "[Python Zen](https://www.python.org/dev/peps/pep-0020)" as much as possible, especially `Explicit is better than implicit`. And for me, 
```
if l:
``` 
is very ambiguous (`l` can be not None, not 0, not False, etc.) whereas:
```
if len(l) > 0:
```
clearly indicates that `l` is an iterable (for example, `len(None)` will trigger an Exception that could be interesting to know. This is the reason why I am using it, and want to keep using it (since there is no performance impact).

File: gef.py [hunk 2]
```diff
@@ -4735,11 +4735,11 @@ def ps(self):
             fields = line.split()
             t = {}
 
-            for i in range(len(names)):
+            for i, name in enumerate(names):
```
Comment (hugsy) [inline gef.py]: 12 years doing Python code, and I am just learning `enumerate()`, I feel like a noob, it's awesome. I like that, very Pythonic.

**Comment by Grazfather** [comment]
actually I disagree. `len(None)` might be one way to check, but to me it's no better than having it fail later. There are a few places where you do redundant checks when one will do.

```python
if l:
```

to me _isn't_ ambiguous, it's just covering all the bases. I read it as, "If I can use `l`". Usually it's just a quick glance at the next use to see what type it is. `len(l)` might tell you it's iterable, but you still need to determine if it's a tuple, list, string, set, or dict, so I don't see how it's any better.

Of course, this is your codebase, so the coding style is up to you.

**Comment by Grazfather** [comment]
As for speed, the difference is probably negligible, but `if len(l) > 0` is two function calls while `bool` is just one. _Maybe_ if this check were in a loop we could see a different between the two, but the real reason is for clarity, and if you find the old way more clear that's your prerogative to keep it that way :)

**Comment by hugsy** [comment]
Hi @Grazfather,

Ok so I ran some `timeit` tests, and it just so happens that your approach (`if l:`) is actually faster by ~15-20%, so I've decided to accept merge all your changes "as is" as part of issue #68 since I am really wanting to make gef faster (I've been using it a  lot on a QEMU MIPS VM lately and I could totally feel the slowliness :disappointed: ).

Thanks!

**Comment by Grazfather** [comment]
Oh that's a better improvement than I expected! Glad these things are adding up to make nice and quick.

**Pull request merged**